examples/Example14/src/Example14EventAction.cc

```diff
@@ -0,0 +1,113 @@
+// * use  in  resulting  scientific  publications,  and indicate your *
+// * acceptance of all terms of the Geant4 Software license.          *
+// ********************************************************************
+//
+#include "Example14EventAction.hh"
+#include "Example14Hit.hh"
+#include "Example14DetectorConstruction.hh"
+
+#include "G4SDManager.hh"
+#include "G4HCofThisEvent.hh"
+#include "G4Event.hh"
+#include "G4EventManager.hh"
+
+#include "G4GlobalFastSimulationManager.hh"
+#include "AdeptIntegration.h"
+
+Example14EventAction::Example14EventAction(Example14DetectorConstruction *aDetector)
+    : G4UserEventAction(), fHitCollectionID(-1), fTimer()
+{
+}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
+
+Example14EventAction::~Example14EventAction() {}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
+
+void Example14EventAction::BeginOfEventAction(const G4Event *)
+{
+  fTimer.Start();
+  
+  // zero the counters
+  number_electrons = 0;
+  number_positrons = 0;
+  number_gammas = 0;
+
+}
+
+//....oooOO0OOooo........oooOO0OOooo........oooOO0OOooo........oooOO0OOooo......
+
+void Example14EventAction::EndOfEventAction(const G4Event *aEvent)
+{
+
+  fTimer.Stop();
+  
+  // Get hits collection ID (only once)
+  if(fHitCollectionID == -1)
+  {
+    fHitCollectionID = G4SDManager::GetSDMpointer()->GetCollectionID("hits");
+  }
+  // Get hits collection
+  auto hitsCollection = static_cast<Example14HitsCollection*>(
+    aEvent->GetHCofThisEvent()->GetHC(fHitCollectionID));
+
+  if(hitsCollection == nullptr)
+  {
+    G4ExceptionDescription msg;
+    msg << "Cannot access hitsCollection ID " << fHitCollectionID;
+    G4Exception("Example14EventAction::GetHitsCollection()", "MyCode0001",
+                FatalException, msg);
+  }
+ 
+  Example14Hit* hit    = nullptr;
+  G4double hitEn       = 0;
+  G4double totalEnergy = 0;
+  auto eventId         = G4EventManager::GetEventManager()->GetConstCurrentEvent()->GetEventID();
+
+  // print number of secondares std::setw(24) << std::fixed
+  std::cout << "EndOfEventAction " << eventId <<  ": electrons " << number_electrons << std::endl;
+  std::cout << "EndOfEventAction " << eventId <<  ": positrons " << number_positrons << std::endl;
+  std::cout << "EndOfEventAction " << eventId <<  ": gammas    " << number_gammas << std::endl;
+ 
+  for(size_t iHit = 0; iHit < hitsCollection->entries(); iHit++)
+  {
+    hit     = static_cast<Example14Hit*>(hitsCollection->GetHit(iHit));
+    hitEn   = hit->GetEdep();
+
+    const char *vol_name = vecgeom::GeoManager::Instance().FindPlacedVolume(iHit)->GetLogicalVolume()->GetName();
+    const char *type = (hit->GetType()==1) ? "AdePT" : "G4";
+    if (hitEn)
+      std::cout << "EndOfEventAction " << eventId << " : id " << std::setw(5) << iHit
+                << "  edep " << std::setprecision(2) << std::setw(12) << std::fixed << hitEn/MeV << " [MeV] logical " << vol_name << std::endl;
```

Comment:
This is still very verbose, in particular when running a decent number of events

Reply:
Fixed now, verbosity for event actions can be changed via: /example14/event/verbose, default is 0, 1 prints just total energy deposit per event, >2 prints hits per placed volume